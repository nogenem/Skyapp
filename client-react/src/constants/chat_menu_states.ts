export const MENU_STATES = {
  MAIN: 1,
  CHANGING_LANGUAGE: 2,
  CHANGING_USER_STATUS: 3,
  CHANGING_USER_THOUGHTS: 4,
} as const;
type TMenuStates = typeof MENU_STATES[keyof typeof MENU_STATES];

export type { TMenuStates };
