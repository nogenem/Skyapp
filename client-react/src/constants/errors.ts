export const INVALID_EMAIL = "Invalid email";
export const CANT_BE_BLANK = "Can't be blank";
export const PASSWORDS_MUST_MATCH = "Passwords must match";
export const fieldIsTooShort = (n: number) =>
  `This field must be at least ${n} chars long`;
