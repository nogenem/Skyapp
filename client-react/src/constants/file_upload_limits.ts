export default {
  fileSize: 5 * 1024 * 1024, // 5MB in bytes
  files: 5,
};
