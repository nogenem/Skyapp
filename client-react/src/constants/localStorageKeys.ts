export const LOCAL_STORAGE_TOKEN = 'skyapp-jwt';
export const LOCAL_STORAGE_THEME_MODE = 'skyapp-theme-mode';
