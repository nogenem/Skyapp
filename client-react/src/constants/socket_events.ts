export const SOCKET_CONNECT = 'connect';
export const SOCKET_DISCONNECT = 'disconnect';

export const IO_SIGNIN = 'AUTH:SIGNIN';
export const IO_SIGNOUT = 'AUTH:SIGNOUT';
export const IO_NEW_USER = 'AUTH:NEW_USER';

export const IO_GET_INITIAL_DATA = 'CHAT:GET_INITIAL_DATA';
export const IO_PRIVATE_CHANNEL_CREATED = 'CHAT:PRIVATE_CHANNEL_CREATED';
export const IO_GROUP_CHANNEL_CREATED = 'CHAT:GROUP_CHANNEL_CREATED';
export const IO_REMOVED_FROM_GROUP_CHANNEL = 'CHAT:REMOVED_FROM_GROUP_CHANNEL';
export const IO_GROUP_CHANNEL_UPDATED = 'CHAT:GROUP_CHANNEL_UPDATED';
export const IO_MESSAGES_RECEIVED = 'CHAT:MESSAGES_RECEIVED';
export const IO_SET_ACTIVE_CHANNEL = 'CHAT:SET_ACTIVE_CHANNEL';
export const IO_SET_LAST_SEEN = 'CHAT:SET_LAST_SEEN';
export const IO_USER_STATUS_CHANGED = 'CHAT:USER_STATUS_CHANGED';
export const IO_USER_THOUGHTS_CHANGED = 'CHAT:USER_THOUGHTS_CHANGED';
