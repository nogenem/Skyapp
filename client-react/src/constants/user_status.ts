export const USER_STATUS = {
  ACTIVE: 1,
  AWAY: 2,
  DO_NOT_DISTURB: 3,
  INVISIBLE: 4,
} as const;

export const USER_STATUS_2_TEXT = {
  [USER_STATUS.ACTIVE]: 'Active',
  [USER_STATUS.AWAY]: 'Away',
  [USER_STATUS.DO_NOT_DISTURB]: 'Do not disturb',
  [USER_STATUS.INVISIBLE]: 'Invisible',
} as const;

type TUserStatus = typeof USER_STATUS[keyof typeof USER_STATUS];

export type { TUserStatus };
