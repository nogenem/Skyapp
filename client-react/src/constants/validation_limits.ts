export const MIN_PASSWORD_LEN = 6;
