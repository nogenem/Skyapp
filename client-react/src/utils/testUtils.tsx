import React from 'react';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { IAppState } from '~/redux/store';

export const getMockStore = () => {
  const middlewares = [thunk];
  return configureStore(middlewares);
};

export const getRenderWithRedux = () => {
  const emptyState: Partial<IAppState> = {};
  const mockStore = getMockStore();

  return (
    ui: React.ReactNode,
    initialState: Partial<IAppState> = emptyState,
  ) => {
    const store = mockStore(initialState);
    return {
      ...render(<Provider store={store}>{ui}</Provider>),
    };
  };
};
