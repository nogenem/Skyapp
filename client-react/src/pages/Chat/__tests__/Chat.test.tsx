import React from 'react';

import { USER_STATUS } from '~/constants/user_status';
import { getRenderWithRedux } from '~/utils/testUtils';

import {
  Chat as ChatPage,
  UnconnectedChat as UnconnectedChatPage,
} from '../index';

const renderWithRedux = getRenderWithRedux();

describe('Connected ChatPage', () => {
  it('renders correctly', () => {
    const { container } = renderWithRedux(<ChatPage />);

    expect(container).toMatchSnapshot();
  });
});

describe('Unconnected ChatPage', () => {
  it('renders CTA when `isUserEmailConfirmed` is false', () => {
    const { getByText } = renderWithRedux(
      <UnconnectedChatPage
        isUserEmailConfirmed={false}
        activeChannel={undefined}
        loggedUserStatus={USER_STATUS.ACTIVE}
        emitUserStatusChanged={() => {}}
      />,
    );

    expect(getByText(/confirm your email/i)).toBeInTheDocument();
  });

  it("doesn't render CTA when `isUserEmailConfirmed` is true", () => {
    const { queryByText } = renderWithRedux(
      <UnconnectedChatPage
        isUserEmailConfirmed
        activeChannel={undefined}
        loggedUserStatus={USER_STATUS.ACTIVE}
        emitUserStatusChanged={() => {}}
      />,
    );

    expect(queryByText(/confirm your email/i)).not.toBeInTheDocument();
  });
});
