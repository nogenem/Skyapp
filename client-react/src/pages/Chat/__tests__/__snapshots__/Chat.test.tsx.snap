// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected ChatPage renders correctly 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <div
      class="MuiPaper-root makeStyles-container-3  MuiPaper-elevation8"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-container-5 MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="makeStyles-container-7"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-8"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
            </svg>
          </div>
          <span
            class="makeStyles-dotContainer-9"
          >
            <span
              class="makeStyles-dot-10 makeStyles-dot_active-11 "
              title="Common:Active"
            />
          </span>
        </div>
        <div
          class="makeStyles-textContainer-6"
        >
          <span
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit MuiTypography-noWrap"
            title=""
          />
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="makeStyles-rightContainer-2"
    >
      <div
        class="makeStyles-content-25"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation8 MuiPaper-rounded"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Messages:Confirm your email
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root makeStyles-cardContent-26"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Please, follow the instructions in the email that we sent to you.
              <br />
              That is a necessary step for you to be able to use our chat system.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
